dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -421,6 +431,46 @@ public Response checkElasticSearch() {
     }
 
+    @GET
+    @Timed
+    @Path("/health")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Successful response if the health checks succeed", description = "Successful response if the health checks succeed, NO authentication")
+    @ApiOperation(value = "Successful response if the health checks succeed", notes = "NO authentication", response = List.class)
+    @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "All health checks successful", content = @Content(mediaType = MediaType.APPLICATION_JSON, array = @ArraySchema(schema = @Schema(implementation = HealthCheckResult.class))))
+    @ApiResponse(responseCode = HttpStatus.SC_BAD_REQUEST + "", description = "Bad Request")
+    @ApiResponse(responseCode = HttpStatus.SC_INTERNAL_SERVER_ERROR + "", description = "Health checks failed")
+    public Set<HealthCheckResult> checkHealth(
+            @Parameter(name = "include", description = "List of health checks to run. If unspecified, run all health checks", in = ParameterIn.QUERY,
+                    array = @ArraySchema(schema = @Schema(implementation = String.class, allowableValues = {"hibernate", "deadlocks", "connectionPool"})))
+            @ApiParam(name = "include", value = "List of health checks to run. If unspecified, run all health checks", allowableValues = "hibernate,deadlocks,connectionPool",
+                    type = "array") @QueryParam(value = "include") List<String> include) {
+        Map<String, HealthCheck.Result> results;
+        boolean allHealthy;
+        if (include.isEmpty()) { // Run all health checks
+            results = this.healthCheckRegistry.runHealthChecks();
+        } else {
+            List<String> invalidNames = include.stream().filter(name -> !healthCheckRegistry.getNames().contains(name)).collect(Collectors.toList());
+            if (!invalidNames.isEmpty()) {
+                String invalidNamesMessage = invalidNames.stream().map(name -> String.format("'%s'", name)).collect(Collectors.joining(", "));
+                String errorMessage = String.format("Could not run health checks. The following health checks don't exist: %s", invalidNamesMessage);
+                LOG.error(errorMessage);
+                throw new CustomWebApplicationException(errorMessage, HttpStatus.SC_BAD_REQUEST);
+            }
+            results = include.stream().collect(Collectors.toMap(name -> name, name -> healthCheckRegistry.runHealthCheck(name)));
+        }
+
+        allHealthy = results.values().stream().allMatch(HealthCheck.Result::isHealthy);
+        if (allHealthy) {
+            return results.entrySet().stream().map(result -> new HealthCheckResult(result.getKey(), result.getValue().isHealthy())).collect(Collectors.toSet());
+        } else {
+            results.entrySet().stream().filter(result -> !result.getValue().isHealthy()).forEach(result -> LOG.error("Health check '{}' failed with error: {}", result.getKey(), result.getValue().getMessage()));
```

Comment:
My personal preference is streaming across many lines:

```
results.entrySet().stream()
   .filter(result -> !result.getValue().isHealthy())
   .forEach(result -> LOG.error("Health check '{}' failed with error: {}", result.getKey(), result.getValue().getMessage()));
```

That way my mind can process the filtering and logic separately, instead of having them combined into one block.

As I commented in another PR, it might be worth coming to agreement on a style; I don't think Google code style covers this, and I know not everybody is into these streams -- it took me a while, but now I'm a convert.

Reply:
I streamed across multiple lines now so hopefully it's more readable, but I agree, we should come to a style agreement. I generally like to use streams if I'm filtering, mapping, collecting, or reducing to a single value

Reply:
FWIW, I slightly prefer the second approach (multiple lines) but don't really feel a need to figure out to encode it into one of our checkstyle/google style plugins. I feel like if the check was more complicated than `isHealthy` than sometimes I will use the first approach, but it is a bit of a judgement call.